final_task/rss_reader/Handler.py

```diff
@@ -0,0 +1,69 @@
+import json
+import logging
+
+import feedparser
+from Entry import Entry
+from Logging import logging_decorator
+
+
+class Handler:
+    """class for handling different options: --version, --json, --limit Limit"""
+    @logging_decorator
+    def __init__(self, source: str, limit: int, version: float):
+        self.source = source
+        self.limit = limit
+        self.version = version
+        self.parsed = feedparser.parse(self.source)
+        logging.info("Handler object created")
+
+    # options of command line:
+    @logging_decorator
+    def option_version(self) -> None:
+        print("version ", self.version)
+
+    @logging_decorator
+    def option_json(self) -> None:
+        for num, entry in enumerate(self.gen_entries()):
+            self.print_to_json(self.convert_to_dict(entry))
+            if num == self.limit - 1:
```

Comment:
Insted of checking the limit on each iteration, you can define the number of iteration to be no more than a limit. For example:
```
for item in item_list[:limit]:
 ...
```
In this case you won't need `break` and `if`

Reply:
Same above

Reply:
In my case method gen_entries() is a generator and I can't use the following construction:
`
for item in item_list[:limit]:
`
Can I use the following construction:
`
for entry in islice(self.gen_entries(), 0, self.limit):
`
instead of your example?

Reply:
I'm not sure if you really need generator in here:)